kube-controllers/pkg/cache/cache.go

```diff
@@ -126,7 +126,9 @@ func (c *calicoCache) Set(key string, newObj interface{}) {
 	if reflect.TypeOf(newObj) != c.ObjectType {
 		c.log.Fatalf("Wrong object type received to store in cache. Expected: %s, Found: %s", c.ObjectType, reflect.TypeOf(newObj))
 	}
-
+	// lock the cache
```

Comment:
Something isn't quite adding up to me here - the underlying map is within the `threadSafeCache` instance, which IIUC should already handle this. 

Do you know why we need this second mutex?